src/sx-design/src/ErrorBoundary/ErrorBoundary.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import React, { type Node } from 'react';
+import sx from '@adeira/sx';
+import { fbt } from 'fbt';
+
+import Heading from '../Heading/Heading';
+import Section from '../Section';
+
+type Props = {|
+  +children: Node,
+  +title?: FbtWithoutString,
+  +code?: string,
+  +onComponentDidCatch?: (Error, { componentStack: string, ... }) => void,
+|};
+
+type State = {|
+  +hasError: boolean,
+|};
+
+export default class ErrorBoundary extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(): State {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  componentDidCatch(error: Error, errorInfo: { componentStack: string, ... }): void {
+    if (this.props.onComponentDidCatch != null) {
+      // log the error to an error reporting service
+      this.props.onComponentDidCatch(error, errorInfo);
+    }
+  }
+
+  render(): Node {
+    if (this.state.hasError) {
+      return (
+        <div className={styles('error')}>
+          <div>
+            <Heading xstyle={styles.h1}>{this.props.code ?? '5XX'}</Heading>
+            <Section xstyle={styles.desc}>
+              <Heading xstyle={styles.h2}>
+                {this.props.title ??
+                  fbt('An unexpected error has occurred.', 'generic error message')}
```

Comment:
What about also displaying a retry button, and allow the consumer to pass in an `onRetry` callback? by default it could do `location.reload` maybe?